include/nbl/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.h

```diff
@@ -475,27 +476,28 @@ class CMaterialCompilerGLSLBackendCommon
 		#include "nbl/nblpack.h"
 			struct prefetch_instr_t
 			{
-				prefetch_instr_t() : qword{ 0ull, 0ull } {}
+				prefetch_instr_t() : qword{ 0ull, 0ull }{}
 
 				_NBL_STATIC_INLINE_CONSTEXPR uint32_t DWORD4_DST_REG_SHIFT = 0u;
 				_NBL_STATIC_INLINE_CONSTEXPR uint32_t DWORD4_DST_REG_WIDTH = 8u;
 				_NBL_STATIC_INLINE_CONSTEXPR uint32_t DWORD4_REG_CNT_SHIFT = DWORD4_DST_REG_SHIFT + DWORD4_DST_REG_WIDTH;
 				_NBL_STATIC_INLINE_CONSTEXPR uint32_t DWORD4_REG_CNT_WIDTH = 2u;
 
-				inline void setDstReg(uint32_t r) { dword4 = core::bitfieldInsert(dword4, r, DWORD4_DST_REG_SHIFT, DWORD4_DST_REG_WIDTH); }
-				inline void setRegCnt(uint32_t c) { dword4 = core::bitfieldInsert(dword4, c, DWORD4_REG_CNT_SHIFT, DWORD4_REG_CNT_WIDTH); }
+				inline void setDstReg(uint32_t r) { s.dword4 = core::bitfieldInsert(s.dword4, r, DWORD4_DST_REG_SHIFT, DWORD4_DST_REG_WIDTH); }
+				inline void setRegCnt(uint32_t c) { s.dword4 = core::bitfieldInsert(s.dword4, c, DWORD4_REG_CNT_SHIFT, DWORD4_REG_CNT_WIDTH); }
 
-				inline uint32_t getDstReg() const { return core::bitfieldExtract(dword4, DWORD4_DST_REG_SHIFT, DWORD4_DST_REG_WIDTH); }
-				inline uint32_t getRegCnt() const { return core::bitfieldExtract(dword4, DWORD4_REG_CNT_SHIFT, DWORD4_REG_CNT_WIDTH); }
+				inline uint32_t getDstReg() const { return core::bitfieldExtract(s.dword4, DWORD4_DST_REG_SHIFT, DWORD4_DST_REG_WIDTH); }
+				inline uint32_t getRegCnt() const { return core::bitfieldExtract(s.dword4, DWORD4_REG_CNT_SHIFT, DWORD4_REG_CNT_WIDTH); }
 
-				union {
+				union{
 					uint64_t qword[2];
 					uint32_t dword[4];
-					struct {
+					struct{
 						STextureData tex_data;
 						uint32_t dword4;
-					} PACK_STRUCT;
```

Comment:
@Crisspl wont this break the sizeof the struct?

Reply:
there was an assert statement few lines below, checking if the size of the struct is equal to some value; It did compile, so probably its okay

Reply:
assert or static assert?

Also does it check the size of both structs?

Reply:
`static_assert(sizeof(prefetch_instr_t) == sizeof_uvec4);`

Reply:
> @Crisspl wont this break the sizeof the struct?

it might, compiler and options dependent

Reply:
@Crisspl just give us the correct asserts for size and alignment then